src/pages/Pages.js

```diff
@@ -1,11 +1,23 @@
 import { Route, Switch } from 'react-router-dom';
-import Main from '../components/main';
+
+import Dashboard from './dashboard';
+import Login from '../pages/login';
 import Tasklist from '../components/tasklist';
 
-export const Pages = () => (
+export const Pages = ({user}) => (
     <Switch>
-        <Route path="/tasklist" component={ Tasklist } key="tasklist"/>
-        <Route path="/" exact component={ Main } key="home"/>
+        {console.log(user)}
+        {
+            user ? [
+                <Route path="/tasklist" component={ Tasklist } key="tasklist"/>,
+                <Route path="/"
+                       render={() => <Dashboard user={user}/>}
+                       key="dashboard"/>
+            ] : [
+                <Route path="/" exact component={ Login } key="login"/>
```

Comment:
why here do you have `exact` attribute and do not set it for ather route with the same path (13 line)?